jmclient/jmclient/wallet_utils.py

```diff
@@ -502,11 +504,11 @@ def get_addr_status(addr_path, utxos, utxos_enabled, is_new, is_internal):
             addr_balance += utxodata['value']
 
             #TODO it is a failure of abstraction here that
-            # the bitcoin core interface is used directly
-            #the function should either be removed or added to bci
-            #or possibly add some kind of `gettransaction` function
+            # the no-history interface is excluded in this way.
```

Comment:
I'm not sure this edit is in the spirit of the comment that @chris-belcher originally added here? His point is about encapsulation/design/architecture of code here, e.g. we try to prevent any access to `Wallet` directly, but only ever via `WalletService` for a similar reason.

I think this case is a little less serious than that though; whether we add a 'am I able to do this kind of tx categorization?' function to our blockchainterface, or actually query the type of the blockchain interface, is fairly minor.

Still, I don't think the change you're suggesting here quite makes sense to me. The nature of the NoHistory interface is such that it can't query transactions, so it's not a 'failure of abstraction that the no-history interface is excluded' imo.

Reply:
I am not totally sure myself. When checking for a specific implementation at least a "failure of abstraction" cannot be completely ruled out. ; -)

Please excuse these small – sometimes silly – change requests. If it were within my capabilities, I wish I could make the whole comment obsolete by proving a better solution. What do you think would be the best wording for the comment here?